### PR TITLE
clippy: Fix a bunch of warnings in `script`

### PR DESCRIPTION
**Pull request by mrobinson**
This is just a portion of the errors that are remaining to be fixed.

---

- [x] `./mach build -d` does not report any errors
- [x] `./mach test-tidy` does not report any errors
- [x] These changes are part of #31575.
- [x] These changes do not require tests because they do not change behavior.

### CODE REVIEW

File: components/script/dom/headers.rs [hunk 1]
```diff
@@ -535,6 +535,7 @@ pub fn extract_mime_type(headers: &HyperHeaders) -> Option<Vec<u8>> {
                 // Step 6.4
                 if temp_essence != essence {
                     charset = temp_charset.map(|c| c.to_string());
+                    temp_essence.clone_into(&mut essence);
```
Comment (gterzian) [inline components/script/dom/headers.rs:538]: Should this not replace the below?

**Review by gterzian — APPROVED** [review]
Thank you! LGTM with onne question.

File: components/script/dom/headers.rs (hunk 1 above)
Comment (mrobinson) [reply components/script/dom/headers.rs:538]: Oh, nice catch. This was an oversight on my part which should be fixed now.

**Comment by mrobinson** [comment]
@gterzian Thanks for the review!

**Pull request merged**